etc/download.sh

```diff
@@ -1,12 +1,40 @@
-#/bin/sh
+#/usr/bin/env bash
 #
 # This script expects a single argument, which should be an URL pointing to a
 # file for download; it then tries to download that file, into a local file
 # with the same name as the remote file.
+# If a file of the same name exists in ~/.gap-package-cache, then this file
+# is used if it has the same hash
 set -e
 
 # check whether curl is available, and if so, delegate to it
-command -v curl >/dev/null 2>&1 && exec curl -L -O "$@"
+command -v curl >/dev/null 2>&1 &&
+{
+    name=${@##*/}
+    # Check if file already exists and has right hash
+    if [ -e ~/.gap-package-cache/${name} ] ; then
+        command -v sha256sum > /dev/null 2>&1 &&
+        {
```

Comment:
Could we perhaps put this code into a separate script, say `etc/ci-download.sh`? Because it is also executed by users which execute a `make bootstrap-pkg-full` and I don't feel comfortable stuffing hundred of MB of data into a directory the user may not be aware exists.

They might also not want to retry; and they might be on a slow connection and not be happy if we abort the download after 10 minutes (download might take 12 minutes on their system...)

Reply:
I'm happy to move things around. My intention is that if you don't have a directory ~/.gap-package-cache none of this stuff will do anything, but that might be too subtle.